example/lib/custom_marker.dart

```diff
@@ -0,0 +1,267 @@
+                )
+            )
+          ]
+      ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: (){
+          //_measurePerformance();
+
+          // Generate random markers
+          var param = <LatLng>[];
+          for (var i = 0; i < RANDOM_MARKER_NUM; i++) {
+            final lat = _rnd.nextDouble() * 20 + 30;
+            final lng = _rnd.nextDouble() * 20 + 125;
+            param.add(LatLng(lat, lng));
+          }
+
+          _mapController.toScreenLocationBatch(param).then((value) {
+            for (var i = 0; i < RANDOM_MARKER_NUM; i++) {
+              var point = Point<double>(value[i].x, value[i].y);
+              _addMarker(point, param[i]);
+            }
+          });
+        },
+        child: Icon(Icons.add),
+      ),
+    );
+  }
+
+  // ignore: unused_element
+  void _measurePerformance() {
+    final trial = 10;
+    final batches = [500, 1000, 1500, 2000, 2500, 3000];
+    var results = Map<int, List<double>>();
+    for (final batch in batches) {
+      results[batch] = [0.0, 0.0];
+    }
+
+    _mapController.toScreenLocation(LatLng(0, 0));
+    Stopwatch sw = Stopwatch();
+
+    for (final batch in batches) {
+      //
+      // primitive
+      //
+      for (var i = 0; i < trial; i++) {
+        sw.start();
+        var list = <Future<Point<num>>>[];
+        for (var j = 0; j < batch; j++) {
+          var p = _mapController.toScreenLocation(LatLng(j.toDouble() % 80, j.toDouble() % 300));
+          list.add(p);
+        }
+        Future.wait(list);
+        sw.stop();
+        results[batch][0] += sw.elapsedMilliseconds;
+        sw.reset();
+      }
+
+      //
+      // batch
+      //
+      for (var i = 0; i < trial; i++) {
+        sw.start();
+        var param = <LatLng>[];
+        for (var j = 0; j < batch; j++) {
+          param.add(LatLng(j.toDouble() % 80, j.toDouble() % 300));
+        }
+        Future.wait([_mapController.toScreenLocationBatch(param)]);
+        sw.stop();
+        results[batch][1] += sw.elapsedMilliseconds;
+        sw.reset();
+      }
+
+      print('batch=$batch,primitive=${results[batch][0] / trial}ms, batch=${results[batch][1] / trial}ms');
+    }
+
+  }
+}
+
+class Marker extends StatefulWidget {
+  final Point _initialPosition;
+  final LatLng _coordinate;
+  final void Function(PositionChangeListener) _addPositionChangeListener;
+
+  Marker(String key, this._coordinate, this._initialPosition, this._addPositionChangeListener) : super(key: Key(key));
+
+  @override
+  State<StatefulWidget> createState() {
+    final state = _MarkerState(_coordinate, _initialPosition);
+    _addPositionChangeListener(state);
+    return state;
+  }
+}
+
+class _MarkerState extends State with TickerProviderStateMixin, PositionChangeListener {
+  final _iconSize = 20.0;
+
+  final LatLng _coordinate;
+  Point _position;
+
+  AnimationController _controller;
+  Animation<double> _animation;
+
+  _MarkerState(this._coordinate, this._position);
```

Comment:
No need to pass in the coordinates. You can just get them from the widget directly.

Reply:
Thanks, fixed